src/components/MenteeDashboard/tracks/TrackEnroll.js

```diff
@@ -0,0 +1,150 @@
+import React, { useState, useEffect } from 'react';
+import { Steps, Button, message, Modal, Radio } from 'antd';
+import { connect } from 'react-redux';
+import { getTracksAction } from '../../../state/tracks/tracksActionCreator';
+import TrackCard from './TracksEnrollCard';
+import CustomLoader from '../../common/Spinner/CustomLoader';
+import tempCourseLogo from '../../assets/image/dashboard/science_image.png';
+import TrackEnrollStyled from './TrackEnrollStyled';
+import TracksEnrollStages from './TracksEnrollStages';
+import { Popconfirm } from 'antd';
+import { QuestionCircleOutlined } from '@ant-design/icons';
+import codeClanApi from '../../../api/apiUtils';
+import EnrollmentStatus from './EnrollmentStatus';
+
+// import TrackEnrollCard from './TracksEnrollCard';
+const { Step } = Steps;
+
+const steps = [
+  {
+    title: 'Tracks',
+    content: 'First-content',
+  },
+  {
+    title: 'Stages',
+    content: 'Second-content',
+  },
+  {
+    title: 'Confirmation',
+    content: 'Last-content',
+  },
+];
+
+function TrackEnroll({
+  visible,
+  getTracksAction,
+  loading,
+  error,
+  errResponse,
+  data,
+  onCancel,
+}) {
+  const [current, setCurrent] = useState(0);
+
+  const [trackId, setTrackId] = useState(null);
+  const [enrollStatus, setEnrollStatus] = useState(null);
+
+  function next() {
+    const newCurrent = current + 1;
+    setCurrent(newCurrent);
+  }
+
+  function prev() {
+    const newCurrent = current - 1;
+    setCurrent(newCurrent);
+  }
+
+  const handleSetTrackId = e => {
+    setTrackId(e.target.value);
+  };
+
+  const userEnrollTrackAction = async id => {
+    try {
+      const res = codeClanApi.post(`/tracks/${id}/enroll`);
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  const handleEnrollTrack = async id => {
+    await userEnrollTrackAction(id);
+    next();
+  };
+  const { items } = data;
+
+  useEffect(() => {
+    getTracksAction();
+  }, []);
```

Comment:
Codacy found an issue: [React Hook useEffect has a missing dependency: 'getTracksAction'. Either include it or remove the dependency array. If 'getTracksAction' changes too often, find the parent component that defines it and wrap that definition in useCallback.](https://app.codacy.com/gh/codeclannigeria/codeclannigeria-frontend/pullRequest?prid=5838854)